src/types.rs

```diff
@@ -399,16 +383,22 @@ impl EventData {
     where
         S: AsRef<str>,
     {
+        let metadata = vec![
+            ("type".to_owned(), event_type.as_ref().to_owned()),
+            ("content-type".to_owned(), "application/json".to_owned()),
+        ]
+        .into_iter()
+        .collect();
```

Comment:
Too much allocation for what we are trying to do. Why not just using a hash map from the get-go?

Reply:
this notation to mimic hashmap literals is recommended by the std libary's hashmap documentation [here](https://doc.rust-lang.org/nightly/std/collections/struct.HashMap.html) and should be fairly optimized.  However, it is a simple change to go back to HashMap::new() and inserting.